dashboard/test/lib/pd/payment/payment_calculator_test.rb

```diff
@@ -0,0 +1,44 @@
+require 'test_helper'
+
+module Pd::Payment
+  class PaymentCalculatorTest < ActiveSupport::TestCase
+    self.use_transactional_test_case = true
+
+    setup_all do
+      @regional_partner = create :regional_partner
+
+      @csf_workshop = create(:pd_ended_workshop, course: Pd::Workshop::COURSE_CSF, enrolled_and_attending_users: 20)
+      @csp_workshop = create(:pd_ended_workshop, course: Pd::Workshop::COURSE_CSP, subject: Pd::Workshop::SUBJECT_CSP_WORKSHOP_1, enrolled_and_attending_users: 20)
+
+      @csf_workshop.enrollments.map(&:user).each do |user|
+        PaymentCalculator::MINIMUM_PUZZLES_FOR_CSF_QUALIFICATION.times do
+          create(:user_level, user: user, best_result: ActivityConstants::PASSING_VALUES.sample)
+        end
+      end
+    end
+
+    test 'Raise error if workshop is not ended' do
+      workshop = create(:pd_workshop, course: Pd::Workshop::COURSE_CSF, enrolled_and_attending_users: 20)
```

Comment:
We probably don't need the overhead (not that it's very much, granted) of creating 20 users here when the condition this is checking is the first line in the function and users are never inspected.